### DISCUSSION TITLE
Certain aggrid events seem to not work?

### DISCUSSION THREAD

**Discussion by BlankAdventure** [zauberzeug/nicegui#2878] [Q&A] [answered]
### Question

From the example code here https://nicegui.io/documentation/aggrid#respond_to_an_ag_grid_event, it says that "All AG Grid events are passed through to NiceGUI via the AG Grid global listener. These events can be subscribed to using the .on() method."

However, rowClicked and rowDoubleClicked specifically do not seem to work; rowSelected does work.

Try each of rowClicked, rowDoubleClicked, rowSelected in the on() method below:

```python
from nicegui import ui

ui.aggrid({
    'columnDefs': [
        {'headerName': 'Name', 'field': 'name'},
        {'headerName': 'Age', 'field': 'age'},
    ],
    'rowData': [
        {'name': 'Alice', 'age': 18},
        {'name': 'Bob', 'age': 21},
        {'name': 'Carol', 'age': 42},
    ],
}).on('rowDoubleClicked', lambda event: ui.notify(f'Cell value: {event.args["value"]}'))

ui.run()
```

**Comment by falkoschindler** [comment, marked as answer]
Hi @BlankAdventure,

> All AG Grid events are passed through to NiceGUI via the AG Grid global listener. These events can be subscribed to using the .on() method.

This statement is correct, even though some event arguments cause trouble when being serialized, because they include cyclic refereces. When double-clicking a row you can see a JavaScript error on the developer console in your browser.

To inspect the event arguments, you can replace the event registration with the following JavaScript handler:
```py
.on('rowDoubleClicked', js_handler='(event) => console.log(event)')
```
or simply
```py
.on('rowDoubleClicked', js_handler='console.log')
```

Assuming we are interested in the "data" and "rowIndex" fields, we can limit the event arguments to these two:
```py
.on('rowDoubleClicked', lambda event: ui.notify(f'Cell value: {event.args}'), ['data', 'rowIndex'])
```

This selection can be serialized without any problems and is correctly displayed in the notification.

![Screenshot 2024-04-12 at 15 02 49](https://github.com/zauberzeug/nicegui/assets/5767091/207164c8-ead1-4ed6-9e75-4403c70bbe98)

**Reply by BlankAdventure** [reply]
Thanks, that was very helpful.

(As a side note I had just opted to use cellDoubleClicked, which highlights the row and also returns the rowId, so effectively the same behaviour)
